storage/backend.go

```diff
@@ -22,6 +22,7 @@ type Backend interface {
 	Upload(ctx context.Context, file io.Reader, path string) error
 	Size(ctx context.Context, path string) (int64, error)
 	Delete(ctx context.Context, path string) error
+	IsObjectExists(ctx context.Context, objectName string) (bool, error)
```

Comment:
What about having a method `Info` which return a custom error if it does not exist?

(I'm not a fan of the `IsObjectExists` function name --'